inference/core/workflows/execution_engine/introspection/blocks_loader.py

```diff
@@ -52,8 +52,33 @@ def describe_available_blocks(
         + dynamic_blocks
     )
     result = []
+    uql_defs = {}
     for block in blocks:
+        if (
```

Comment:
I am voting for explicit metadata for each block manifest, similar to metadata we add to display icons in UI

* we could manifest `"requires_uql": True`
* we can add any context-related metadata that would trigger specific reactions in UI, let's just sit and agree on what is needed